compiler/ml/translcore.ml

```diff
@@ -49,6 +49,31 @@ let transl_extension_constructor env path ext =
 
 (* Translation of primitives *)
 
+let translate_unified_ops (prim : Primitive.description) (env : Env.t)
+    (lhs_type : type_expr) : Lambda.primitive option =
+  (* lhs_type is already unified in type-level *)
+  let entry = Hashtbl.find_opt Unified_ops.index_by_name prim.prim_name in
+  match entry with
+  | Some {specialization} -> (
+    match specialization with
+    | {int}
```

Comment:
This always takes precedence int over other types. (Rule 1-2) The last default case is a fallback strategy. (Rule 3)

Reply:
Can you give an example of any change after removing this first case?
All the other cases seem to have incompatible when clauses, so it would fall back to the last case no matter what.
Unless I'm missing something.

Reply:
You are right. There is no logical difference in the behavior after removing that case. I was just thinking of the computational difference. In the existing codebase, I assume the first case is hit the most frequently (since it was originally int-only).

Reply:
It's just a detail anyway. Not much difference. Whatever you think is best.